docs/README.md

```diff
@@ -0,0 +1,137 @@
+| [local](http-challenge/local.md) | uses a local path at your server/machine for validation/hash files
+
+### dns-challenge
+
+Following providers for a dns-challenge are currently supported.
+
+| Provider                                | Description
+|-----------------------------------------|-------------
+| [AutoDNS](dns-challenge/autodns.md)     |
+| [Azure](dns-challenge/azure.md)         |
+| [hetzner](dns-challenge/hetzner.md)     |
+| [openstack](dns-challenge/openstack.md) |
+| [pebble](dns-challenge/pebble.md)       | used for integration tests in github workflow
+
+## General variables
+
+| Variable                            | Required | Default | Description
+|-------------------------------------|----------|---------|------------
+| **domain configuration**
+| certificate_name                    | yes      |         | name of the resulting certificate. Most useful for wildcard certificates to not have files named '*.example.com' on the filesystem
+| zone                                | yes      |         | zone in which the dns records should be created
+| subject_alt_name                    | yes      |         | if you want to use wildcard-certificates use base name again as otherwise DNS txt record creation could fail
+| email_address                       | yes      |         | mail address which is used for the certificate (reminder mails are sent here)
+| **configuration options**           |          |         |
+| account_key_content                 | no       |         | content of the created letsencrypt account key
+| private_key_content                 | no       |         | content of the created private key for the certificate (allows reuse of keys)
+| letsencrypt_do_http_challenge       | yes      | false   | use http challenge
+| letsencrypt_do_dns_challenge        | yes      | false   | use dns challenge
+| letsencrypt_use_acme_live_directory | no       | false   | choose if production certificates should be created, the staging directory of LE will be used by default
+| force_renewal                       | no       |         | Force renewal of certificate before `remaining_days` is reached
+
+## Variables for http-challenge
+
+| Variable                            | Required | Default | Description
+|-------------------------------------|----------|---------|------------
+| letsencrypt_http_provider           | yes      |         | which http provider should be used: s3, local
+
+## Variables for dns-challenge
+
+| Variable                            | Required | Default | Description
+|-------------------------------------|----------|---------|------------
+| dns_user                            | yes      |         | username to access the DNS api
+| dns_password                        | yes      |         | password to access the DNS api
+| letsencrypt_dns_provider            | yes      |         | which DNS provider should be used: autodns, azure, hetzner, pebble, openstack
```

Comment:
I don't like the mixing of description and list of providers, this will get inconsistent very soon.

Reply:
I removed the listing and added a hint to the provider documentation